.github/workflows/ci.yaml

```diff
@@ -84,20 +85,33 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
 
-      # since disk-size is currently too-low, we will build only the docker images required for our purposes
+      # since github-actions gives us 14G only, and fill it up with some garbage
+      # we will free up some space for us (~2GB)
+      - name: Freeing some disk space
+        run: docker system prune --all --force
+
       - name: Build
-        run: |
-          make controller dashboard handler-builder-golang-onbuild
+        run: make docker-images
         env:
           NUCLIO_DOCKER_REPO: ${{ env.REPO }}/${{ env.REPO_NAME }}
 
+      # Clean build leftovers
+      - name: Post build cleanup
+        run: docker system prune --force
+
       - name: Save
         run: |
-          docker save \
-            ${REPO}/${REPO_NAME}/dashboard:${NUCLIO_LABEL}-amd64 \
-            ${REPO}/${REPO_NAME}/controller:${NUCLIO_LABEL}-amd64 \
-            ${REPO}/${REPO_NAME}/handler-builder-golang-onbuild:${NUCLIO_LABEL}-amd64-alpine \
-            | gzip > nuclio_docker_images.tar.gz
+          # list built images
+          # join with " "
+          # save docker images
+          # compress using pigz (use fast, the difference is insignificant)
+          docker images \
+              --filter="reference=${REPO}/${REPO_NAME}/*:${NUCLIO_LABEL}*" \
+              --format "{{ .Repository }}:{{ .Tag }}" \
+            | paste -sd " " - \
+            | docker save $(</dev/stdin) \
```

Comment:
The reason we used a single `docker save X Y Z` (the current) and not `docker save X`, `docker save Y`, `docker save Z` (the suggested) as xargs will execute, is that we
1. redirecting to pigz in order to compress it (faster!)
2. would want a single `.tar.gz` containing those images (rather than multiple compressed files)

Reply:
That's a single docker save and will result in one targz